### ISSUE TITLE
Allow to redirect logs to file or syslog or both

### ISSUE THREAD

**Issue by foxcpp** [emersion/maddy#26]
Global directive?
```
log <targets...>
```

```
log stderr
```
Write to stderr (default).

```
log file_name
```
Write to file.

```
log syslog
```
Send messages to local syslog daemon.

Can be combined:
```
log stderr syslog /var/log/maddy/maddy.log
```

**Comment by foxcpp** [comment]
Should we handle some signal (perhaps SIGUSR1 or SIGUSR2) as a request to reopen the log file? This will allow the logrotate daemon to not restart maddy to correctly perform the rotation.

**Comment by emersion** [comment]
>Should we handle some signal

Not a fan of this, we'll end up with more features we want than signals available. Any idea how other daemons handle this?

Maybe people who want this should use stdout + a thin wrapper process that does what you said.

**Comment by foxcpp** [comment]
>Not a fan of this, we'll end up with more features we want than signals available. Any idea how other daemons handle this?

Other daemons either have full-blown IPC interface or offer only one-two basic functions like "reload" config using SIGUSR1/SIGUSR2 signals.

Speaking of available signals, since Linux 2.2 we have 31 signals for application use (SIGRTMIN-SIGRTMAX, excluding two possibly used by glibc internally (let's say we use sqlite3).

**Comment by emersion** [comment]
>"reload config" using SIGUSR1/SIGUSR2 signals

I'm okay with that.

>Speaking of available signals, since Linux 2.2 we have 31 signals for application use

This is not POSIX though.
